src/functional/ffront/func_to_foast.py

```diff
@@ -240,12 +203,9 @@ def visit_AnnAssign(self, node: ast.AnnAssign, **kwargs) -> foast.Assign:
             assert isinstance(
                 node.annotation, ast.Constant
             ), "Annotations should be ast.Constant(string). Use StringifyAnnotationsPass"
-            globalns = {**fbuiltins.BUILTINS, **self.captured_vars.globals}
-            localns = self.captured_vars.nonlocals
-            annotation = eval(node.annotation.value, globalns, localns)
-            target_type = symbol_makers.make_symbol_type_from_typing(
-                annotation, globalns=globalns, localns=localns
-            )
+            globalns = {**fbuiltins.BUILTINS, **self.closure_vars}
```

Comment:
```suggestion
            ns = {**fbuiltins.BUILTINS, **self.closure_vars}
```
It is fine to use the `globalns` parameter below, but the variable here does represent the global namespace, but the merged global + nonlocal one.